frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -419,32 +418,54 @@ export const fetchCardData = createThunkAction(
   },
 );
 
-export const fetchDashboardCardData = createThunkAction(
-  FETCH_DASHBOARD_CARD_DATA,
-  options => (dispatch, getState) => {
+const fetchDashboardCardDataSync = createAction(FETCH_DASHBOARD_CARD_DATA);
+
+export const fetchDashboardCardData =
+  ({ skipOngoingDashcardRequests, ...options }) =>
+  (dispatch, getState) => {
     const dashboard = getDashboardComplete(getState());
     const selectedTabId = getSelectedTabId(getState());
-    const dashcardIds = [];
-    const promises = getCurrentTabDashboardCards(dashboard, selectedTabId)
+
+    const nonVirtualDashcardsToFetch = getCurrentTabDashboardCards(
+      dashboard,
+      selectedTabId,
+    )
       .filter(({ dashcard }) => !isVirtualDashCard(dashcard))
-      .map(({ card, dashcard }) => {
-        dashcardIds.push(dashcard.id);
-        return dispatch(fetchCardData(card, dashcard, options)).then(() => {
-          return dispatch(updateLoadingTitle());
-        });
+      .filter(({ dashcard }) => {
+        if (skipOngoingDashcardRequests) {
+          const loadingIds = getLoadingDashCards(getState()).loadingIds;
+          return !loadingIds.includes(dashcard.id);
+        }
+
+        return true;
       });
 
-    dispatch(setDocumentTitle(t`0/${promises.length} loaded`));
+    const newLoadingIds = nonVirtualDashcardsToFetch.map(({ dashcard }) => {
+      return dashcard.id;
+    });
+    dispatch(
+      fetchDashboardCardDataSync({
```

Comment:
I was gonna do that, but thought it isn't a pattern we use. Now you mentioned this. I found 80 or so places where we use this style, so it could be good to go.